packages/components/src/Filters/FilterInput.tsx

```diff
@@ -0,0 +1,52 @@
+import React, { useState, useEffect } from 'react';
+import { Checkbox, Radio } from '@patternfly/react-core';
+
+export interface FilterInputProps {
+  addRemoveFilters?: (value?: string, param?: string, type?: string, checked?: boolean) => void;
+  className?: string;
+  currentPage?: string;
+  id: string;
+  label: React.ReactNode;
+  param?: string;
+  value?: string;
+  filters?: any;
+  type?: string;
+}
+
+const FilterInput: React.FC<FilterInputProps> = ({ addRemoveFilters, param = '', filters = {}, type = 'checkbox', value, id, label }) => {
+  const [checked, setChecked] = useState(false);
+  useEffect(() => {
+    if (type !== 'radio') {
+      if (param in filters && filters?.[param]?.includes(value)) {
+        setChecked(true);
+      } else {
+        setChecked(false);
+      }
+    }
+  }, []);
+  const handleChange = () => {
+    switch (type) {
+      case 'checkbox':
+        setChecked(!checked);
+        break;
+    }
```

Comment:
Previously it was a `switch` which got under our radar. I think it would be awesome to change it to `if` since it just checks for one `type`.